commands/operator-sdk/cmd/test/local.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdtest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	cmdError "github.com/operator-framework/operator-sdk/commands/operator-sdk/error"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+
+	"github.com/spf13/cobra"
+)
+
+type testLocalConfig struct {
+	kubeconfig        string
+	globalManPath     string
+	namespacedManPath string
+	goTestFlags       string
+	namespace         string
+}
+
+var tlConfig testLocalConfig
+var (
+	defaultDirFileMode = 0750
+	defaultFileMode    = 0644
+)
+
+func NewTestLocalCmd() *cobra.Command {
+	testCmd := &cobra.Command{
+		Use:   "local <path to tests directory> [flags]",
+		Short: "Run End-To-End tests locally",
+		Run:   testLocalFunc,
+	}
+	defaultKubeConfig := ""
+	homedir, ok := os.LookupEnv("HOME")
+	if ok {
+		defaultKubeConfig = homedir + "/.kube/config"
+	}
+	testCmd.Flags().StringVar(&tlConfig.kubeconfig, "kubeconfig", defaultKubeConfig, "Kubeconfig path")
+	testCmd.Flags().StringVar(&tlConfig.globalManPath, "global-manifest", "", "Path to manifest for Global resources (e.g. CRD manifests)")
+	testCmd.Flags().StringVar(&tlConfig.namespacedManPath, "namespaced-manifest", "", "Path to manifest for per-test, namespaced resources (e.g. RBAC and Operator manifest)")
+	testCmd.Flags().StringVar(&tlConfig.goTestFlags, "go-test-flags", "", "Additional flags to pass to go test")
+	testCmd.Flags().StringVar(&tlConfig.namespace, "namespace", "", "If non-empty, single namespace to run tests in")
+
+	return testCmd
+}
+
+func testLocalFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("operator-sdk test local requires exactly 1 argument"))
```

Comment:
remove cmdError.
We should remove this util from the refactor branch altogether to avoid using it.